django/contrib/postgres/forms/hstore.py

```diff
@@ -23,6 +23,8 @@ def prepare_value(self, value):
     def to_python(self, value):
         if not value:
             return {}
+        if isinstance(value, dict):
+            value = json.dumps(value)
         try:
             value = json.loads(value)
         except ValueError:
```

Comment:
Since everything is forced to `six.text_type` below I think it would be safe to skip the serialization for `dict` values.

e.g.

```python
if not isintance(value, dict):
    try:
        value = json.loads(value)
    except ValueError:
        ...
```